### PR TITLE
Enable Sphinx spellcheck for docs generation

### LINKED ISSUE #10264: Add Sphinx Spell-Checker
We have been fixing various typos in the project but it would be good if we can enable a Spell Checker for our Docsite. So our docs are typo-free.

We can use https://pypi.org/project/sphinxcontrib-spelling/ to do this. 
**Docs**: https://sphinxcontrib-spelling.readthedocs.io/en/latest/

I gave it a shot but unfortunately, there are too many words that we need to add in `docs/spelling_wordlist.txt`.

Here is the list I had but there are still ~6k more words that need to be added, if someone wants to give it a shot:

```
Acyclic
Airbnb
Async
Avro
Bas
BaseView
Cassanda
DagRun
Dask
Dataproc
Datastore
Gantt
Gunicorn
Harenslak
Hashicorp
Jarek
Jinja
Jira
Kamil
Kerberos
Kibana
Kubernetes
Oozie
Opsgenie
Parameterizing
Potiuk
Py
Qubole
Sqoop
Standarization
Systemd
Templating
XCom
XComs
Zsh
adls
airflow
airflowignore
ansible
apikey
argcomplete
args
async
auth
autocommit
autodetect
automl
autoscale
aws
backend
backfill
backfilled
bashcompinit
batcher
bigquery
bigtable
bitshift
boto
botocore
catchup
cfg
chown
classmethod
cloudant
cloudsql
cncf
config
configMapRef
configmap
configuing
cronjob
crypto
cyexamplekey
dag
dagbag
dagruns
databricks
datadog
dataset
datasets
datetime
dbs
dejson
deserializing
dest
dev
devel
dingding
distros
dockerenv
docstring
docstrings
elasticsearch
envFrom
eventlet
exampleinclude
exasol
facebook
failover
fernet
fluentd
fs
gRPC
gcp
gcpcloudsql
gevent
github
greenlets
grpc
gssapi
hadoop
hashicorp
hdfs
hiveserver
howto
httpbin
imap
initdb
integration
integrations
jalr
jdbc
jinja
keytab
krb
kubernetes
kwargs
kylin
licence
literalinclude
logins
loglevel
logstash
lshift
macOS
mdeng
memorystore
mesos
metadatabase
metarouter
metastore
mongo
msg
mssql
noqa
odbc
papermill
param
paramiko
petabyte
pgdatabase
pghost
pgpassfile
pgpassword
pgport
pguser
pidfile
pinot
postgre
postgres
postgresql
precheck
proc
programmatically
psql
py
pylint
pythonpath
rankdir
rbac
readthedocs
resetdb
rshift
rst
salesforce
saml
sanitization
searchpath
secretRef
secretsmanager
seealso
serverless
sftp
smtps
spegno
sqla
stackdriver
statsd
stdout
subcommand
subdag
subgraph
subpackage
subpackages
subprocesses
sudo
tablename
templated
templating
teradata
timedelta
umask
unpause
upgradedb
upsert
uptime
utcnow
versionable
vertica
wasb
webhdfs
webserver
xcom
xxxxxxxx
yandex
yandexcloud

```

### PR DESCRIPTION
**Pull request by caddac**
closes: #10264

Runs spell check during docs build. All misspelled/dictionary words are in `docs/spelling_wordlist.txt`
I've spent some effort to comb the list and remove (and update in the docs) the actual
misspelled words.

### CODE REVIEW

**Comment by kaxil** [comment]
@potiuk Do we need to bump the EPOCH number so that the CI uses the updated Dockerfile ?

**Comment by potiuk** [comment]
> @potiuk Do we need to bump the EPOCH number so that the CI uses the updated Dockerfile ?

Absolutely not. It's only needed in case stuff is removed)

**Comment by potiuk** [comment]
And only for pip installs.

**Comment by kaxil** [comment]
> > @potiuk Do we need to bump the EPOCH number so that the CI uses the updated Dockerfile ?
> 
> Absolutely not. It's only needed in case stuff is removed)

Any ideas on how we fix the failing test:

https://github.com/apache/airflow/pull/10280/checks?check_run_id=969045276#step:5:6023
```
THESE PACKAGES DO NOT MATCH THE HASHES FROM THE REQUIREMENTS FILE. If you have updated the package versions, please update the hashes. Otherwise, examine the package contents carefully; someone may have tampered with them.
```

**Comment by potiuk** [comment]
I re-run it. It looks like a transient error. Since we are modifying the dependencie pretty early in the Dockerfile, It gets rebuild from the point it is modified and sometimes there is a problem when downloading the packages where checksum does not match on some transient 404 not found error (usually this happens when apt get gets an error during download but this error is somehow swallowed by firewall or smth).  And since we are rebuilding the image in all steps that need it, this might happen rather frequently.  You will notice that in this PR each "Build CI image" takes much longer than usual 3-4 minutes (it might be 12 minutes or more because almost all image is rebuilt).


BTW. This is one of the problems my (soon to submit) PR is going to address. We are going to build each image only once - so even if Dockerfile is modified at the beginning the build it will be built once and such pre-built image will be used by all the subsequent steps.

**Comment by kaxil** [comment]
Got it, I restarted the tests again 🤞

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: This actually is a typo.

It is used in `privacy_notice.rst` -> It should be `browser`

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: This is a typo used in `CHANGELOG.txt` -> it should be just `cloud`

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: This all are spelling errors

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt:579]: legit typo. I think this is captured from a commit SHA

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: genuine typo

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: genuine typo

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: genuine typo

File: docs/spelling_wordlist.txt
Comment (caddac) [inline docs/spelling_wordlist.txt]: Jobs will not work if the bicket is not specified. --> `ticket`?

File: docs/spelling_wordlist.txt
Comment (caddac) [reply docs/spelling_wordlist.txt]: `commiter` is used in the string `<li><p>[AIRFLOW-XXXX] Fix typo commiter =&gt; committer (#7392)</p></li>`. thoughts on how to handle that?

File: docs/spelling_wordlist.txt
Comment (caddac) [inline docs/spelling_wordlist.txt:808]: thoughts on alternate spellings like this?

File: docs/spelling_wordlist.txt
Comment (caddac) [inline docs/spelling_wordlist.txt:813]: used in `- [AIRFLOW-397] Documentation: Fix typo "instatiating" to "instantiating"`

File: docs/spelling_wordlist.txt
Comment (caddac) [inline docs/spelling_wordlist.txt]: used in `- [AIRFLOW-1714] Fix misspelling: s/seperate/separate/`

File: docs/spelling_wordlist.txt
Comment (caddac) [reply docs/spelling_wordlist.txt:813]: I went ahead and changed this

File: docs/spelling_wordlist.txt
Comment (caddac) [reply docs/spelling_wordlist.txt]: I went ahead and changed this

File: docs/spelling_wordlist.txt
Comment (kaxil) [reply docs/spelling_wordlist.txt]: I think it is actually `bucket`

File: docs/spelling_wordlist.txt
Comment (kaxil) [reply docs/spelling_wordlist.txt:808]: I think that should be allowed, we can standardize it later if we need to but for the sake of this PR I think we should allow this

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: Replace the occurrences of `weren't` to `were not`

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo in faq.rst --> should be values

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo... should just be You in datacatlog.rst

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo, correct one is `sqlite`

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: Since this is fixed, we can remove this, right?

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo, should be `pformat`

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo needs fixing --> `mismatch`

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: minimum

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: Switch

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: Qubole

File: docs/spelling_wordlist.txt
Comment (kaxil) [inline docs/spelling_wordlist.txt]: typo.,.. `Metastore`

**Review by kaxil — COMMENTED** [review]
Happy to merge this once the comment I left are fixed/addressed

File: docs/spelling_wordlist.txt
Comment (caddac) [reply docs/spelling_wordlist.txt]: that makes way more sense

**Comment by caddac** [comment]
I've fixed all those spelling mistakes and rebased.  Thanks!

**Review by kaxil — APPROVED** [review]
Great work, thanks @caddac

**Comment by kaxil** [comment]
😬 tests failed with 5 typos:

```
_api/airflow/providers/snowflake/transfers/snowflake_to_slack/index.rst:14:specifiying::results_df }}'. The 'results_df' variable name can be changed by specifiying a different
changelog.rst:66:pformat::Use pformat instead of str to render arguments in WebUI (#9587)
_api/airflow/providers/google/cloud/hooks/kubernetes_engine/index.rst:92:weren::resources) will not be deleted if they weren’t present at the
_api/airflow/providers/exasol/hooks/exasol/index.rst:ExaConnection::(optional) passed into pyexasol.ExaConnection.export_to_pandas method
_api/airflow/providers/apache/hive/hooks/hive/index.rst:DataFrame::(optional) passed into pandas.DataFrame constructor

```

**Comment by caddac** [comment]
Ya, I did that on purpose ;). Should be good now.

**Comment by caddac** [comment]
@kaxil All set here! Thanks for your help!

**Comment by potiuk** [comment]
Wohooo! :tada: !

File: UPDATING.md
Comment (ashb) [inline UPDATING.md:647]: This shouldn't have been changed. (Not super important, but `AIRFLOW-6392` is correct.)

**Pull request merged**
